File: device-app/android/Java/helloWorld/app/src/main/java/com/rationalowl/hello/MyApp.java
package com.rationalowl.hello;

import android.app.Application;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;

import com.google.firebase.FirebaseApp;
import com.rationalowl.minerva.client.android.MinervaManager;
import com.rationalowl.minerva.client.android.util.Logger;

public class MyApp extends Application {
    private static final String TAG = "MyApp";


    public void onCreate(){
        Logger.debug(TAG, "onCreate enter");
        super.onCreate();
        Context context = getApplicationContext();

        // initialize fcm
        FirebaseApp.initializeApp(context);

        // initialize rational owl
        MinervaManager.init(context);

        // set rationalowl message listener
        MinervaManager minMgr = MinervaManager.getInstance();
        minMgr.setMsgListener(new RoMessageListener());

        // register sample channel
        NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        NotificationChannel channel = new NotificationChannel("sampleChannelId","sample fcm Channel", NotificationManager.IMPORTANCE_DEFAULT);
        notificationManager.createNotificationChannel(channel);
    }
}

File: device-app/android/Kotlin/helloWorld_240120/app/src/main/java/com/rationalowl/hello/MyApp.kt
package com.rationalowl.hello

import android.app.Application
import android.app.NotificationChannel
import android.app.NotificationManager
import com.google.firebase.FirebaseApp
import com.rationalowl.minerva.client.android.MinervaManager
import com.rationalowl.minerva.client.android.util.Logger

class MyApp : Application() {
    override fun onCreate() {
        Logger.debug(TAG, "onCreate enter")
        super.onCreate()
        val context = applicationContext

        // initialize fcm
        FirebaseApp.initializeApp(context)

        // initialize rational owl
        MinervaManager.init(context)

        // set rationalowl message listener
        val minMgr = MinervaManager.getInstance()
        minMgr.setMsgListener(RoMessageListener())

        // register sample channel
        val notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
        val channel = NotificationChannel(
            "sampleChannelId",
            "sampleChannel",
            NotificationManager.IMPORTANCE_DEFAULT
        )
        notificationManager.createNotificationChannel(channel)
    }

    companion object {
        private const val TAG = "MyApp"
    }
}